src/NUnitCommon/nunit.agent.core/AgentOptions.cs

```diff
@@ -0,0 +1,125 @@
+
+                    if (!VALID_OPTIONS.ContainsKey(opt))
+                        throw new Exception($"Invalid argument: {arg}");
+
+                    bool optionTakesValue = VALID_OPTIONS[opt];
+
+                    if (optionTakesValue)
+                    {
+                        if (val == null && index + 1 < args.Length)
+                            val = args[++index];
+
+                        if (val == null)
+                            throw new Exception($"Option requires a value: {arg}");
+                    }
+                    else if (delim > 0)
+                    {
+                        throw new Exception($"Option does not take a value: {arg}");
+                    }
+
+                    if (opt == "agentId")
+                        AgentId = new Guid(GetArgumentValue(arg));
+                    else if (opt == "agencyUrl")
+                        AgencyUrl = GetArgumentValue(arg);
+                    else if (opt == "debug-agent")
+                        DebugAgent = true;
+                    else if (opt == "debug-tests")
+                        DebugTests = true;
+                    else if (opt == "trace")
+                        TraceLevel = (InternalTraceLevel)Enum.Parse(typeof(InternalTraceLevel), val.ShouldNotBeNull());
+                    else if (opt == "pid")
+                        AgencyPid = val.ShouldNotBeNull();
+                    else if (opt == "work")
+                        WorkDirectory = val.ShouldNotBeNull();
+                    else
+                        throw new Exception($"Invalid argument: {arg}");
+                }
+                else if (File.Exists(arg))
+                    Files.Add(arg);
+                else
+                    throw new FileNotFoundException($"FileNotFound: {arg}");
+            }
+
+            if (Files.Count > 1)
+                throw new ArgumentException($"Only one file argument is allowed but {Files.Count} were supplied");
+
+            string GetArgumentValue(string argument)
+            {
+                var delim = argument.IndexOfAny(DELIMS);
+
+                if (delim > 0)
+                    return argument.Substring(delim + 1);
+
+                if (index + 1 < args.Length)
+                    return args[++index];
+
+                throw new Exception($"Option requires a value: {argument}");
+            }
+        }
+
+        public Guid AgentId { get; } = Guid.Empty;
+        public string AgencyUrl { get; } = string.Empty;
+        public string AgencyPid { get; } = string.Empty;
+        public bool DebugTests { get; } = false;
+        public bool DebugAgent { get; } = false;
+        public InternalTraceLevel TraceLevel { get; } = InternalTraceLevel.Off;
+        public string WorkDirectory { get; } = string.Empty;
+
+        public List<string> Files { get; } = new List<string>();
+
+        private static bool IsOption(string arg)
+        {
+            return arg.StartsWith("--");
```

Comment:
nitpick: I think this `StartsWith` overload will be culture-sensitive by default. Could we pass a `StringComparison` in?
 
```suggestion
            return arg.StartsWith("--", StringComparison.Ordinal);
```